python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -444,7 +462,8 @@ def __new__(cls, downstream_asset_key: AssetKey, input_name: str):
 @whitelist_for_serdes
 class ExternalAssetNode(
     namedtuple(
-        "_ExternalAssetNode", "asset_key dependencies depended_by op_name op_description job_names"
+        "_ExternalAssetNode",
+        "asset_key dependencies depended_by op_name op_description job_names partitions_def",
```

Comment:
Someone skimming this code might see an attr named `partitions_def` and think that all `PartitionsDefinition` subtypes are supported, when in fact I believe only `StaticPartitionsDefinitions` are supported, and all other subtypes are not.

A more typesafe way of handling this could be to define an abstract base class called something like `ExternalPartitionsDefinition`, which gets extended by `ExternalTimeWindowPartitionsDefinition` and `ExternalStaticPartitionsDefinition`.  And then have `ExternalAssetNode` have an attr of type `ExternalTimeWindowPartitionsDefinition`.

Reply:
Yes, this makes sense. I'll add the abstract base class